ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,1103 @@
+                if (currentLevelNumber === null) {
+                    breadcrumbsList.push([0, (chart.get(e.newRootId) as SeriesType)]);
+                    currentLevelNumber = 0;
+                } else {
+                    breadcrumbsList.push([currentLevelNumber + 1, (chart.get(e.newRootId) as SeriesType)]);
+                }
+            } else {
+                let node = e.target.nodeMap[e.newRootId];
+                const extraNodes = [];
+
+                // When the root node is set and has parent,
+                // recreate the path from the node tree.
+                while (node.parent || node.parent === '') {
+                    extraNodes.push(node);
+                    node = e.target.nodeMap[node.parent];
+                }
+                extraNodes.reverse().forEach(function (node): void {
+                    if (currentLevelNumber === null) {
+                        breadcrumbsList.push([0, node]);
+                        currentLevelNumber = 0;
+                    } else {
+                        breadcrumbsList.push([++currentLevelNumber, node]);
+                    }
+                });
+            }
+        }
+        if (!this.isTreemap && drilldownLevels && drilldownLevels.length) {
+            // Add the initial series as the first element.
+            if (!breadcrumbsList[0]) {
+                breadcrumbsList.push([null, (drilldownLevels[0].seriesOptions as any)]);
+                currentLevelNumber = -1;
+            }
+
+            drilldownLevels.forEach(function (level): void {
+                // If level is already added to breadcrumbs list,
+                // don't add it again- drilling categories
+                if (level.levelNumber > (currentLevelNumber as number)) {
+                    breadcrumbsList.push([level.levelNumber, (level.pointOptions as Point)]);
+                }
+            });
+        }
+        this.breadcrumbsList = breadcrumbsList;
+    }
+
+    /**
+     * Default button text formatter.
+     *
+     * @requires  modules/breadcrumbs
+     *
+     * @function Highcharts.Breadcrumbs#defaultFormatter
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     * @param {Highcharts.Breadcrumbs} breadcrumb
+     *        Breadcrumb.
+     * @return {string}
+     *         Formatted text.
+     */
+    public defaultFormatter(this: Breadcrumbs, breadcrumb: Array<number|null|Point|SeriesType|SVGAElement>): string {
+        const breadcrumbs = this,
+            chart = breadcrumbs.chart,
+            breadcrumbsOptions = breadcrumbs.options,
+            arrow = breadcrumbsOptions.showFullPath ? '' : '◁ ',
+            lang = chart.options.lang;
+
+        return breadcrumbsOptions.formatter && breadcrumbsOptions.formatter(breadcrumb, breadcrumbs) ||
+            arrow + (
+                format(breadcrumbsOptions.format as string, { point: (breadcrumb[1] as Point) }, chart) ||
```

Comment:
If not sure `point` is the right name, as this may also be a series in drilldown or a level in treemap. Would `level` be a better name?